lib/autoflow.coffee

```diff
@@ -28,6 +28,19 @@ module.exports =
     paragraphs = []
     # Convert all \r\n and \r to \n. The text buffer will normalize them later
     text = text.replace(/\r\n?/g, '\n')
+
+    leadingVerticalSpace = text.match(/^\s*\n/)
+    if leadingVerticalSpace
+      text = text.substr(leadingVerticalSpace.length)
+    else
+      leadingVerticalSpace = ''
+
+    trailingVerticalSpace = text.match(/\n\s*$/)
+    if trailingVerticalSpace
+      text = text.substr(0, text.length - trailingVerticalSpace.length)
+    else
+      trailingVerticalSpace = ''
+
     paragraphBlocks = text.split(/\n\s*\n/g)
```

Comment:
For the record: this change feels clumsy to me.  It's a bit too verbose for the conceptual weight of what's being accomplished.

But it does appear to me to be necessary to fix the problem *at this point*, rather than by pruning the first and last elements of `paragraphBlocks` after the split, or by fixing up the results after the reflow.  The reason is that we want to preserve the selection boundaries, and we don't want to end up merging the first paragraph with what precedes it, or the last with what follows.  The leading and trailing vertical space must still be there after the reflow operation, so saving it and putting it back seems like the right solution.

The core problem is that `reflowSelection` is really only well defined when the selection boundaries do not fall in the whitespace between paragraphs.